src/conversions/either.rs

```diff
@@ -94,7 +94,13 @@ where
         } else if let Ok(r) = obj.extract::<R>() {
             Ok(Either::Right(r))
         } else {
-            let err_msg = format!("failed to convert the value to '{}'", Self::type_input());
+            // TODO: it might be nice to use the `type_input()` name here once `type_input`
+            // is not experimental, rather than the Rust type names.
+            let err_msg = format!(
+                "failed to convert the value to 'Union[{}, {}]'",
+                std::any::type_name::<L>(),
+                std::any::type_name::<R>()
+            );
```

Comment:
I don't totally love degrading the error message here, some folks may find it breaking.

One option is to use the `type_input()` message when that feature is enabled, which would mean folks who were already using this feature would have nothing change. And then have this new code be the `#[cfg(not(feature = "experimental-inspect")]` fallback.

... but then the error message would vary based on features enabled, and I wasn't sure I liked that either 🙈 

Opinions welcome on which is the lesser of the two evils!

Reply:
I think this what we should have done in the first place and any breakage now is bugs happening which is not nice but happens, i.e. let's keep it like this and maybe simplify if the inspection ever looses the feature flag.